src/MSBuild/XMake.cs

```diff
@@ -2595,22 +2598,22 @@ private static void StartLocalNode(CommandLineSwitches commandLineSwitches)
             {
                 Exception nodeException = null;
                 NodeEngineShutdownReason shutdownReason = NodeEngineShutdownReason.Error;
+
                 // normal OOP node case
                 if (nodeModeNumber == 1)
                 {
-                    OutOfProcNode node = new OutOfProcNode();
-
                     // If FEATURE_NODE_REUSE is OFF, just validates that the switch is OK, and always returns False
                     bool nodeReuse = ProcessNodeReuseSwitch(commandLineSwitches[CommandLineSwitches.ParameterizedSwitch.NodeReuse]);
-                    string[] lowPriorityInput = commandLineSwitches[CommandLineSwitches.ParameterizedSwitch.LowPriority];
-                    bool lowpriority = lowPriorityInput.Length > 0 && lowPriorityInput[0].Equals("true");
-
+                    OutOfProcNode node = new OutOfProcNode();
                     shutdownReason = node.Run(nodeReuse, lowpriority, out nodeException);
 
                     FileUtilities.ClearCacheDirectory();
                 }
                 else if (nodeModeNumber == 2)
                 {
+                    // TaskHost nodes don't need to worry about node reuse or low priority. Node reuse is always off, and TaskHosts
```

Comment:
It's kinda deeply baked in. To do this right™, we'd need to change a number of places that gather parameters for the node handshake. Currently, most assume the only relevant ones are architecture and runtime. Ideally, we'd also tweak our public API a little; I'm thinking of an intrinsic function that tells you whether there's a TaskHost currently running that fits your parameters but only allows runtime and architecture.

Where I got a bit bogged down was in OutOfProcTaskHost.cs. LowPriority actually shouldn't be too hard because I can just ask the current process what its priority is, but it doesn't know about whether node reuse is enabled or not, and I had a little trouble passing it information. Note that for this case, it isn't enough to have a node that knows tell it whether it should be shutting down or waiting around for another connection because we'd need to match on the nodeReuse bit.

Honestly, several aspects of this felt like they were working by accident 😥